ocean_lib/ocean/mint_fake_ocean.py

```diff
@@ -26,7 +25,7 @@ def mint_fake_OCEAN(config: Config) -> None:
     with open(addresses_file) as f:
         network_addresses = json.load(f)
 
-    web3 = Web3(provider=get_web3_connection_provider(config.network_url))
+    web3 = get_web3(get_web3_connection_provider(config.network_url))
```

Comment:
It's kinda too bad we have to add yet another layer of indirection. Everyone knows `Web3()`. Makes it a bit harder for anyone reading the code to grok.

Now we have nested calls, one call inside another: `web3 = get_web3(get_web3_connection_provider(network_url))`. 

I see an opportunity! All the changes here that replace 'Web3()` with `get_web3()` has this nested structure, with `get_web3_connection_provider()` as the inner call. 

Therefore, let's collapse the nested structure. That is: make `get_web3()` take `network_url` as the argument rather than provider. Then inside its implementation, the first thing it can do is `provider = get_web3_connection_provider(network_url)`.

Make sense? If yes, please proceed:)